### PR TITLE
Rotation causes search bar to sized incorrectly

### LINKED ISSUE #4466: Rotating while searchController open results in incorrect size
@rachelsquirrel noted that the search bar is fine when one is holding the device in portrait mode. Then when the app is flipped to landscape and then back to portrait, the search bar moves up. Here are their screenshots after they turned their screens to landscape then back to portrait:

I have found out the Site Search also does this

screen-shot-2015-11-11-at-11-53-03

img_3786

### PR DESCRIPTION
**Pull request by kwonye**
Resolves #4466

Fixes it for both the `BlogListViewController` and `PostListViewController`

Repro steps:
1. Open search in portrait
2. Rotate to landscape
3. Rotate back to portrait

Expected:
The search bar is correctly sized in both landscape and portrait mode

Needs review: @aerych

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Blog/BlogListViewController.m [hunk 1]
```diff
@@ -160,15 +160,14 @@ - (void)viewWillDisappear:(BOOL)animated
     self.resultsController.delegate = nil;
 }
 
-- (void)willAnimateRotationToInterfaceOrientation:(UIInterfaceOrientation)toInterfaceOrientation duration:(NSTimeInterval)duration
+- (void)viewWillTransitionToSize:(CGSize)size withTransitionCoordinator:(id<UIViewControllerTransitionCoordinator>)coordinator
 {
-    [super willAnimateRotationToInterfaceOrientation:toInterfaceOrientation duration:duration];
-    if (self.tableView.tableHeaderView == self.headerView) {
-        [self updateHeaderSize];
```
Comment (aerych) [inline WordPress/Classes/ViewRelated/Blog/BlogListViewController.m]: Seems like we'd still need this call.

Comment (kwonye) [inline WordPress/Classes/ViewRelated/Blog/BlogListViewController.m]: Oh my goodness, that's definitely a cut-paste error, thanks! Fixing now.

Comment (kwonye) [inline WordPress/Classes/ViewRelated/Blog/BlogListViewController.m]: Addressed in 7797d8d

**Comment by aerych** [comment]
Hiya @kwonye.  Took this for a quick test drive and the issue with the status bar is no longer overlapping the `UISearchBar`.  However, I noticed that if I dismiss the search bar and rotate again it reappears beneath the nav bar like so: 
simulator screen shot nov 17 2015 2 01 50 pm

I think the check check for `self.searchWrapperViewHeightConstraint.constant > 0` might still be needed.

**Comment by kwonye** [comment]
> I think the check check for self.searchWrapperViewHeightConstraint.constant > 0 might still be needed.

Thanks for that catch @aerych, I thought the view visibility was set to NO when it was hidden but I was wrong. Fixed!

**Comment by aerych** [comment]
Works great :)
:shipit:

**Comment by kwonye** [comment]
Thanks for the review! (Sorry about the mistakes)

**Pull request merged**